custom_components/hacs/base.py

```diff
@@ -926,26 +928,29 @@ async def async_get_category_repositories(self, category: HacsCategory) -> None:
                 self.repositories.mark_default(repository)
                 if self.status.new and self.configuration.dev:
                     # Force update for new installations
-                    self.queue.add(repository.common_update())
+                    tasks.append(repository.common_update())
                 continue
 
-            self.queue.add(
+            tasks.append(
                 self.async_register_repository(
                     repository_full_name=repo,
                     category=category,
                     default=True,
                 )
             )
+        self.queue.add(tasks)
```

Comment:
Where are the tasks in the queue cleaned up if there are remaining tasks on shutdown?

We should probably not add coroutines directly but either coroutine functions or asyncio.Tasks so we avoid running into "coroutine was never awaited".

Reply:
This is a valid point, but I think that should be changed separately as it doesn't really have anything to do with the changes in this PR,